_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_vaapi.cpp

```diff
@@ -570,7 +566,13 @@ mfxStatus SetFrameRate(
         misc_param->type = VAEncMiscParameterTypeFrameRate;
         frameRate_param = (VAEncMiscParameterFrameRate *)misc_param->data;
 
-        frameRate_param->framerate = TL_attached ? base_framerate / extTL.Layer[par.m_numLayers - 1 - tl].FrameRateScale : base_framerate;
```

Comment:
Are you 100% sure that par.m_numLayers always > 1? In case of par.m_numLayers equals to 0, or equals to 1 with tl>1 you'll get invalid index for the array.

Reply:
Should add
if (tl == numTL-1)
     PackMfxFrameRate(nom, denom, frameRate_param->framerate);
else
    Pack... as here
It will work when either isn't extTL or it is.